protocol/message.go

```diff
@@ -210,57 +173,43 @@ func ParseMessage(message []byte) (*Message, error) {
 		return nil, fmt.Errorf("message metadata to have an integer (message-id) as second field, but was %v", meta[1])
 	}
 
-	publishingTime, err := strconv.ParseInt(meta[5], 10, 64)
+	var expiresTime *time.Time
+	if meta[5] != "" {
+		if t, err := time.Parse(time.RFC3339, meta[5]); err != nil {
+			return nil, fmt.Errorf("message metadata expected to have a  time string (expiration time) as sixth field, but was %v: %s", meta[5], err.Error())
+		} else {
+			expiresTime = &t
+		}
+	}
+
+	publishingTime, err := strconv.ParseInt(meta[6], 10, 64)
 	if err != nil {
-		return nil, fmt.Errorf("message metadata to have an integer (publishing time) as sixth field, but was %v", meta[5])
+		return nil, fmt.Errorf("message metadata to have an integer (publishing time) as seventh field, but was %v:", meta[6])
 	}
 
-	nodeID, err := strconv.ParseUint(meta[6], 10, 8)
+	nodeID, err := strconv.ParseUint(meta[7], 10, 8)
 	if err != nil {
-		return nil, fmt.Errorf("message metadata to have an integer (nodeID) as seventh field, but was %v", meta[6])
+		return nil, fmt.Errorf("message metadata to have an integer (nodeID) as eighth field, but was %v", meta[7])
 	}
 
-	msg := &Message{
+	m := &Message{
 		ID:            id,
 		Path:          Path(meta[0]),
 		UserID:        meta[2],
 		ApplicationID: meta[3],
+		Expires:       expiresTime,
```

Comment:
I must disagree on this. Since `Cosmin` already approved this I must consider the field as being ok. Cause I think `Expires` naming is very commonly used.